src/game/Spell.cpp

```diff
@@ -1958,9 +1958,40 @@ void Spell::SetTargetMap(SpellEffectIndex effIndex, uint32 targetMode, UnitList&
         }
         case TARGET_SINGLE_FRIEND:
         case TARGET_SINGLE_FRIEND_2:
-            if (m_targets.getUnitTarget())
-                targetUnitMap.push_back(m_targets.getUnitTarget());
+        {
+            if (Unit* unitTarget = m_targets.getUnitTarget())
+            {
+                if (unitTarget->IsFriendlyTo(m_caster))
+                    targetUnitMap.push_back(unitTarget);
+                else
+                {
```

Comment:
there's no reason for this many [ } in my opinion, especially not since most of them only contain 1 line, just keep this one, along with the one on line 1975 and the associated }, then you can remove the rest in my opinion.